up42/order.py

```diff
@@ -118,68 +196,16 @@ def is_fulfilled(self) -> bool:
         """
         return self.status == "FULFILLED"
 
-    def get_assets(self) -> List[asset.Asset]:
+    def get_assets(self) -> Iterator[asset.Asset]:
         """
         Gets the Order assets or results.
         """
-        if self.is_fulfilled:
-            params: asset_searcher.AssetSearchParams = {"search": self.order_id}
-            assets_response = asset_searcher.search_assets(self.auth, params=params)
-            return [asset.Asset(self.auth, asset_info=asset_info) for asset_info in assets_response]
-        raise ValueError(f"Order {self.order_id} is not FULFILLED! Current status is {self.status}")
-
-    @classmethod
-    def place(cls, auth: up42_auth.Auth, order_parameters: dict, workspace_id: str) -> "Order":
-        """
-        Places an order.
-
-        Args:
-            auth: An authentication object.
-            order_parameters: A dictionary for the order configuration.
-
-        Returns:
-            Order: The placed order.
-        """
-        url = host.endpoint(f"/v2/orders?workspaceId={workspace_id}")
-        response_json = auth.request(
-            request_type="POST",
-            url=url,
-            data=_translate_construct_parameters(order_parameters),
-        )
-        if response_json["errors"]:
-            message = response_json["errors"][0]["message"]
-            raise ValueError(f"Order was not placed: {message}")
-        order_id = response_json["results"][0]["id"]
-        order = cls(auth=auth, order_id=order_id, order_parameters=order_parameters)
-        logger.info("Order %s is now %s.", order.order_id, order.status)
-        return order
-
-    @staticmethod
-    def estimate(auth: up42_auth.Auth, order_parameters: OrderParams) -> int:
-        """
-        Returns an estimation of the cost of an order.
-
-        Args:
-            auth: An authentication object.
-            order_parameters: A dictionary for the order configuration.
-
-        Returns:
-            int: The estimated cost of the order
-        """
-
-        url = host.endpoint("/v2/orders/estimate")
-        response_json = auth.request(
-            request_type="POST",
-            url=url,
-            data=_translate_construct_parameters(order_parameters),
-        )
-        estimated_credits: int = response_json["summary"]["totalCredits"]
-        logger.info(
-            "Order is estimated to cost %s UP42 credits (order_parameters: %s)",
-            estimated_credits,
-            order_parameters,
-        )
-        return estimated_credits
+        if not (self.is_fulfilled) and self.status != "BEING_FULFILLED":
+            raise ValueError(f"Order {self.order_id} is not valid. Current status is {self.status}")
+        params = {"search": self.order_id, "page": 0}
+        for page in self._get_pages(self.session, "/v2/assets", params):
+            for asset_info in page:
+                yield asset.Asset(base.workspace.auth, asset_info=asset_info)
```

Comment:
we can't completely get rid of auth element until we migrate asset class to descriptor.